arbnode/transaction_streamer.go

```diff
@@ -988,14 +1019,64 @@ func (s *TransactionStreamer) WriteMessageFromSequencer(
 		BlockHash:       &msgResult.BlockHash,
 	}
 
+	l1Block := msgWithMeta.Message.Header.BlockNumber
+
+	if s.config().SovereignSequencerEnabled && s.espressoClient != nil && wavmio.IsHotShotLive(l1Block) {
+		ctx, cancel := context.WithTimeout(context.Background(), s.config().EspressoTimeout)
+		defer cancel()
+		err := s.SendL2MessageToEspresso(ctx, msgWithMeta)
+		if err != nil {
+			return fmt.Errorf("failed to send message to espresso: %w", err)
+		}
+	}
+
 	if err := s.writeMessages(pos, []arbostypes.MessageWithMetadataAndBlockHash{msgWithBlockHash}, nil); err != nil {
 		return err
 	}
+
 	s.broadcastMessages([]arbostypes.MessageWithMetadataAndBlockHash{msgWithBlockHash}, pos)
 
 	return nil
 }
 
+func (s *TransactionStreamer) SendL2MessageToEspresso(ctx context.Context, msgWithMeta arbostypes.MessageWithMetadata) error {
+
+	tx := espressoTypes.Transaction{
+		Namespace: s.config().EspressoNamespace,
+		Payload:   msgWithMeta.Message.L2msg,
```

Comment:
I am thinking maybe we can add another `L2MessageKind` and modify the `L2Msg` to this kind. So that the `validators` can handle these messages easily.
https://github.com/EspressoSystems/nitro-espresso-integration/blob/e60c027a33f1b3fc07dbac9afc8429b60f9487d9/arbos/parse_l2.go#L110